extensions/amp-story/0.1/system-layer.js

```diff
@@ -0,0 +1,258 @@
+          <path d="M0 0h24v24H0z" fill="none"/>
+        </svg>
+      </div>
+      <div role="button" class="i-amphtml-story-mute-audio-control i-amphtml-story-button">
+        <svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="#FFFFFF">
+          <path d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"/>
+          <path d="M0 0h24v24H0z" fill="none"/>
+        </svg>
+      </div>
+      <div role="button" class="i-amphtml-story-exit-fullscreen i-amphtml-story-button" hidden>
+        <svg fill="#FFFFFF" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
+          <path d="M0 0h24v24H0z" fill="none"/>
+          <path d="M5 16h3v3h2v-5H5v2zm3-8H5v2h5V5H8v3zm6 11h2v-3h3v-2h-5v5zm2-11V5h-2v5h5V8h-3z"/>
+        </svg>
+      </div>
+      <div div role="button" class="i-amphtml-story-bookend-close i-amphtml-story-button" hidden>
+        <svg fill="#FFFFFF" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
+          <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/>
+          <path d="M0 0h24v24H0z" fill="none"/>
+        </svg>
+      </div role="button">
+    </div>`;
+/*eslint-enable max-len */
+
+
+/**
+ * @param {!Element} el
+ * @param {boolean} isHidden
+ */
+function toggleHiddenAttribute(el, isHidden) {
+  if (isHidden) {
+    el.setAttribute('hidden', 'hidden');
+  } else {
+    el.removeAttribute('hidden');
+  }
+}
+
+
+/**
+ * System Layer (i.e. UI Chrome) for <amp-story>.
+ */
+export class SystemLayer {
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @private {!Window} */
+    this.win_ = win;
+
+    /** @private {boolean} */
+    this.isBuilt_ = false;
+
+    /** @private {?Element} */
+    this.root_ = null;
+
+    /** @private {boolean} */
+    this.inFullScreen_ = false;
+
+    /** @private {?Element} */
+    this.exitFullScreenBtn_ = null;
+
+    /** @private {?Element} */
+    this.closeBookendBtn_ = null;
+
+    /** @private {?Element} */
+    this.muteAudioBtn_ = null;
+
+    /** @private {?Element} */
+    this.unmuteAudioBtn_ = null;
+
+    /** @private @const {!ProgressBar} */
+    this.progressBar_ = ProgressBar.create(win);
+  }
+
+  /**
+   * @param {number} pageCount The number of pages in the story.
+   * @return {!Element}
+   */
+  build(pageCount) {
+    if (this.isBuilt_) {
+      return this.getRoot();
+    }
+
+    this.isBuilt_ = true;
+
+    this.root_ = this.win_.document.createElement('aside');
+    this.root_.classList.add('i-amphtml-story-system-layer');
+    this.root_./*OK*/innerHTML = TEMPLATE;
```

Comment:
"NOTOK" 😛

Reply:
Since this template is just a constant (no publisher-specified text or markup), isn't this XSS safe and thus /*OK*/?

Reply:
Yes, *for now*. It's easier to forbid setting `innerHTML` than guaranteeing its usage will never change and introduce a XSS vector. Also, `createElement` is less error prone in general than a large block of markup.

That being said, I was referencing @dvoytenko in https://github.com/ampproject/amphtml/pull/11526#discussion_r142260487. Happy to leave this as a fast-fix after merge, but we should mark it high priority.

Reply:
Filed #11545 to track.

Reply:
Added a comment about why it's okay, just in case.